test/lit.cfg

```diff
@@ -1648,6 +1648,22 @@ if not getattr(config, 'target_run_simple_swift', None):
         '%%line-directive %%t/main.swift -- '
         '%s %%t/a.out'
         % (config.target_build_swift, mcp_opt, config.target_codesign, config.target_run))
+    # TODO: Remove when forward mode AD support is robust.
+    config.target_run_simple_swift_forward_mode_differentiation = (
+        '%%empty-directory(%%t) && '
+        '%s %%s -enable-experimental-forward-mode-differentiation -o %%t/a.out %s -module-name main  && '
+        '%s %%t/a.out &&'
+        '%s %%t/a.out'
+        % (config.target_build_swift, mcp_opt, config.target_codesign, config.target_run))
```

Comment:
I think we can avoid adding ad-hoc `lit` substitutions by using the `%target-run-simple-swift\(([^)]+)\)` parameterized substitution:

https://github.com/apple/swift/blob/770d43028ec6657ecc005d8e0c81c12da821e674/test/lit.cfg#L1713

```swift
// RUN: %target-run-simple-swift(-enable-experimental-forward-mode-differentiation)
```

We could add a similar substitution for `%target-run-simple-swiftgyb(...)` too!

Reply:
Added `%target-run-simple-swiftgyb(...)` in https://github.com/apple/swift/pull/30918 to enable some other forward-mode differentiation tests.

Reply:
Done.